### PR TITLE
pythonPackages.stack-data: init at 0.0.7

### PR DESCRIPTION
**Pull request by jluttine**
###### Motivation for this change

Add Python package stack-data and its dependencies.

IPython will soon need this dependency: https://github.com/ipython/ipython/blob/master/setup.py#L194 (permalink https://github.com/ipython/ipython/blob/a483781c6e480b18b6cdc78a9851c23cd528a890/setup.py#L194), so with this PR stack-data would be readily available in nixpkgs when IPython release will depend on stack-data. I needed to add this to nixpkgs already now because I was using master branch of IPython locally. So I thought I'd provide a PR for these upcoming dependencies.

I'll run nixpkgs-review soon.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jluttine** [comment]
nixpkgs-review passed

**Comment by jluttine** [comment]
I now rebased on master. While this PR has been waiting for half a year now, someone else had added a few packages that I had in this PR, namely `pure-eval` and `executing`. So, some duplicate work gone to waste. Anyway, two new packages still remain in this PR.

File: pkgs/development/python-modules/stack-data/default.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  };
+
+  nativeBuildInputs = [
+    git
+  ];
+
+  buildInputs = [
+    setuptools_scm
+    toml
+  ];
```
Comment (jonringer) [inline pkgs/development/python-modules/stack-data/default.nix:36]:
```suggestion
  nativeBuildInputs = [
    git
    setuptools_scm
    toml
  ];
```

File: pkgs/development/python-modules/stack-data/default.nix [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  # Install requires .git and fetchFromGitHub doesn't support leaving it so
+  # let's use fetchgit
+  src = fetchgit {
+    url = "https://github.com/alexmojaki/${pname}.git";
+    rev = "v${version}";
+    sha256 = "0smifhq8ma8qqhsawmgbfxq5i07rszgysf23i38dlal4h9krm03z";
+    leaveDotGit = true;
+  };
+
```
Comment (jonringer) [inline pkgs/development/python-modules/stack-data/default.nix:28]:
```suggestion
  };

  SETUPTOOLS_SCM_PRETEND_VERSION = version;
```

File: pkgs/development/python-modules/stack-data/default.nix [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  propagatedBuildInputs = [
+    asttokens
+    executing
+    pure-eval
+  ];
+
+  checkInputs = [
+    littleutils
+    pygments
+    pytest
```
Comment (jonringer) [inline pkgs/development/python-modules/stack-data/default.nix:47]:
```suggestion
    pytestCheckHook
```
tests aren't actually being ran
```
running build_ext

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
Finished executing setuptoolsCheckPhase
pytestcachePhase
/nix/store/v12flc8l02vg93dynbyrp46q99bks7fd-python3.8-stack_data-0.0.7
```

File: pkgs/development/python-modules/littleutils/default.nix [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+buildPythonPackage rec {
+  pname = "littleutils";
+  version = "0.2.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0vwijrylppmk0nbddqvn527r9cg3zw8d6zk6r58hslry42jf7jp6";
+  };
+
```
Comment (jonringer) [inline pkgs/development/python-modules/littleutils/default.nix:14]: There are no tests included. Please try to checkout from source and check if they have unit tests, and try to run them. Unit tests give a good indication that they package has a high degree of validity and correctness given the python package set.

If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors. Please see [pythonImportsCheck documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck) for more information.

```

----------------------------------------------------------------------
Ran 0 tests in 0.000s
```

Comment (jluttine) [reply pkgs/development/python-modules/littleutils/default.nix:14]: Thanks! Yeah, this tiny package doesn't have any tests at all even in the source repository. I explicitly disabled the check phase and used `pythonImportsCheck` now. :+1:

File: pkgs/development/python-modules/stack-data/default.nix (hunk 3 above)
Comment (jluttine) [reply pkgs/development/python-modules/stack-data/default.nix]: Oh, now I learned something new! :+1: Yes, fixed now.

File: pkgs/development/python-modules/stack-data/default.nix (hunk 2 above)
Comment (jluttine) [reply pkgs/development/python-modules/stack-data/default.nix]: Ah, interesting! This `SETUPTOOLS_SCM_PRETEND_VERSION` was new to me, thanks! :+1: Fixed.

File: pkgs/development/python-modules/stack-data/default.nix (hunk 1 above)
Comment (jluttine) [reply pkgs/development/python-modules/stack-data/default.nix]: I'm always very confused if packages should go to `buildInputs` or `nativeBuildInputs`. The confusion is even greater when it seems that the package builds just fine either way.. But I now made this fix. Thanks!

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 88494 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>5 packages built:</summary>  <ul>    <li>python27Packages.littleutils</li>    <li>python37Packages.littleutils</li>    <li>python37Packages.stack-data</li>    <li>python38Packages.littleutils</li>    <li>python38Packages.stack-data</li>  </ul></details>

**Comment by jluttine** [comment]
@jonringer Great, thanks a lot for the review! :ok_hand:

**Comment by jluttine** [comment]
Oh, I realized `pname = "stack_data"` should have been normalized to `pname = "stack-data"` and then `repo = "stack_data"`.. Should I still fix that or leave it as it is?

**Comment by jonringer** [comment]
> Oh, I realized `pname = "stack_data"` should have been normalized to `pname = "stack-data"` and then `repo = "stack_data"`.. Should I still fix that or leave it as it is?

That's more for the "attr" in `python-modules.nix`, the pname can match what's on pypi

**Pull request merged**